config/opal_functions.m4

```diff
@@ -480,24 +480,56 @@ dnl #######################################################################
 # of the assignment in foo=`which <prog>`). This macro ensures that we
 # get a sane executable value.
 AC_DEFUN([OPAL_WHICH],[
-# 1 is the variable name to do "which" on
-# 2 is the variable name to assign the return value to
+    # 1 is the variable name to do "which" on
+    # 2 is the variable name to assign the return value to
+
+    OPAL_VAR_SCOPE_PUSH([opal_prog opal_file opal_dir opal_sentinel])
+
+    opal_prog=$1
+
+    # There are 3 cases:
+
+    # 1. opal_prog is an absolute filename.  If that absolute filename
+    # exists and is executable, return $2 with that name.  Otherwise,
+    # $2 is unchanged.
+
+    # 2. opal_prog is a relative filename (i.e., it contains one or
+    # more /, but does not begin with a /).  If that file exists
+    # relative to where we are right now in the filesystem and is
+    # executable, return the absolute path of that value in $2.
+    # Otherwise, $2 is unchanged.
+
+    # 3. opal_prog contains no /.  Search the PATH for an excutable
+    # with the appropriate name.  If found, return the absolute path
+    # in $2.  Otherwise, $2 is unchanged.
+
+    # Note that these three cases are exactly what which(1) does.
+
+    # Case #3: opal_prog contains no /
+    echo $opal_prog | grep -q /
```

Comment:
Maybe should use DIRECTORY_SEPARATOR instead of the hardcoded /

Reply:
Good point.  Will fix both places.